Snap_rounding_2/include/CGAL/Float_snap_rounding_2.h

```diff
@@ -0,0 +1,830 @@
+    std::vector<std::size_t> updated_polyline;
+    updated_polyline.push_back(poly.front());
+    for(std::size_t i=1; i!=poly.size(); ++i){
+      if(pts[poly[i-1]].x()==pts[poly[i]].x()){
+        Iterator_set_x start, end;
+        // Get all vertices between the two endpoints along x order
+        if(Less_indexes_xy_2(poly[i-1],poly[i])){
+          start=p_sort_by_x.upper_bound(poly[i-1]);
+          end=p_sort_by_x.lower_bound(poly[i]);
+          for(auto it=start; it!=end; ++it){
+            updated_polyline.push_back(*it);
+          }
+        } else {
+          start=p_sort_by_x.upper_bound(poly[i]);
+          end=p_sort_by_x.lower_bound(poly[i-1]);
+          for(auto it=end; it!=start;){
+            --it;
+            updated_polyline.push_back(*it);
+          }
+        }
+      }
+      updated_polyline.push_back(poly[i]);
+    }
+    std::swap(poly, updated_polyline);
+  }
+}
+
+template <class Concurrency_tag=Sequential_tag, class Traits, class PointsRange , class PolylineRange>
+void double_snap_rounding_2_disjoint(PointsRange &pts, PolylineRange &polylines, const Traits &traits)
+{
+  using Point_2   = typename Traits::Point_2;
+  using Construct_round_point_2    = typename Traits::Construct_round_point_2;
+
+  Construct_round_point_2 round = traits.construct_round_point_2_object();
+
+  snap_rounding_scan(pts, polylines, traits);
+#ifdef DOUBLE_2D_SNAP_VERBOSE
+  std::cout << "Round" << std::endl;
+#endif
+  for(Point_2 &p: pts)
+    p=round(p);
+#ifdef DOUBLE_2D_SNAP_VERBOSE
+  std::cout << "Merging points that are collapsed together" << std::endl;
+#endif
+  merge_duplicate_points_in_polylines(pts, polylines, traits);
+#ifdef DOUBLE_2D_SNAP_VERBOSE
+  // The algorithm prevents the a vertex that goes through a segment but a vertex may lie on an horizontal/vertical segments after rounding
+  std::cout << "Subdivide vertical segments with vertices on them" << std::endl;
+#endif
+  snap_post_process(pts, polylines, traits);
+}
+
+} // end of namespace internal
+
+/**
+* ingroup PkgSnapRounding2Ref
+*
+* Given a range of segments, compute rounded subsegments that are pairwise disjoint in their interiors, based on the input curves.
+* The output is a range of polyline with each polyline corresponding to an input segment.
+*
+* TODO Currently compute_subcurves have no visitor to obtain a polyline per input segment, thus
+* currently the output contain one polyline per ubsegment of the arrangement
+*
+* @tparam Concurrency_tag allows choosing whether the algorithm runs in parallel or sequentially.
+* If `CGAL::Parallel_tag` is specified and CGAL is linked with the Intel TBB library, the algorithm will run in parallel.
+* Otherwise, if `CGAL::Sequential_tag` is specified (the default), the algorithm will run sequentially.
+* @tparam InputIterator iterator over a range of `Segment_2`
+* @tparam OutputContainer inserter over a range of `Polyline`. `Polyline` must be a type that provides a `push_back(Point_2)` function.
```

Comment:
missing some doc here. I guess you mean something like *a model of the concepts [RandomAccessContainer](https://doc.cgal.org/latest/Manual/classRandomAccessContainer.html) and [BackInsertionSequence](https://doc.cgal.org/latest/Manual/classBackInsertionSequence.html) whose value type is the so-called `Polyline` type.*